### ISSUE TITLE
✨ Application Revamp: Résumé page

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#6040]
## ✨ Feature

Review your Resume step of the application, where experiences are shown but NOT skills. Slight changes to experience accordions. Null state if no experiences yet.

## 🕵️ Details

This lists your experiences. The content of the accordions is a little different than existing ones on our profile. Each shows something specific to the experience type, and then "Tasks and Responsibilities", which is the experience "details" in the API.

DO NOT show the "How I used x skill" section. Skills should not appear on this page at all.

Experiences can be sorted by date or by type.

## 🎨 Design File

https://www.figma.com/file/BXYVsu1fo9LS2IySJDUINY/Applicants---Job-application?node-id=636-59839&t=uYeh8QDrmOQsNpJQ-0

## 📸 Screenshot

### First time user (ie if now experiences exist yet)
![Step 3 - Resume review (first time user)](https://user-images.githubusercontent.com/4313717/227048278-a59efed6-404a-46c5-ad7a-4748f9540a9d.png)

### Review
![Step 3 - Resume review (complete)](https://user-images.githubusercontent.com/4313717/227048190-fc4f44ec-70d8-43d4-bcd2-2d14dd3125b0.png)

### Work accordion
![image](https://user-images.githubusercontent.com/4313717/227049805-7ffe4d99-085e-4471-bdc5-ce1e4f4fe116.png)

### Education accordion
![image](https://user-images.githubusercontent.com/4313717/227049838-f713f039-7e1e-4325-9064-f0a9951b49c5.png)

### Community accordion
![image](https://user-images.githubusercontent.com/4313717/227049900-f08e661d-f074-41ec-bb91-a4d6735d3e31.png)

### Personal accordion
![image](https://user-images.githubusercontent.com/4313717/227049969-e2fd0407-4206-4a0a-8fce-13fb18a5217d.png)

### Award accordion
![image](https://user-images.githubusercontent.com/4313717/227050015-d77c9cca-02ce-4bf3-94b9-c9b062e88c0f.png)

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

| EN                                                                                                                                                                                                                                                                                                                                                         | FR                                                                                                                                                                                                                                                                                                                                                                                                                              | Description                                                                                                      |
|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------|
| Update   and review your résumé information.                                                                                                                                                                                                                                                                                                               | Mettez à jour et révisez les   renseignements de votre CV.                                                                                                                                                                                                                                                                                                                                                                      | Application step to begin   working on profile, page subtitle                                                    |
| Great   work! On to your résumé.                                                                                                                                                                                                                                                                                                                           | Excellent travail! Passons à   votre CV.                                                                                                                                                                                                                                                                                                                                                                                        | Application step to begin   working on profile, title                                                            |
| The next   step is to make sure that your résumé is as up-to-date as possible.                                                                                                                                                                                                                                                                             | L’étape suivante consiste à vous   assurer que votre CV est aussi à jour que possible.                                                                                                                                                                                                                                                                                                                                          | Application step to begin   working on profile, description, paragraph one                                       |
| More   specifically, not only do we want to know your work history, but we highly   value knowing about your community experience, awards, personal initiatives,   and education too!                                                                                                                                                                      | Plus   précisément, non seulement nous voulons connaître votre parcours   professionnel, mais également votre expérience communautaire, vos   récompenses, vos initiatives personnelles et votre éducation!                                                                                                                                                                                                                     | Application   step to begin working on profile, description, paragraph two                                       |
| Once   you’ve completed your resume and are happy with the experiences you’ve added,   you’ll use them in further steps to help us better understand how you meet   the skill requirements for this opportunity.                                                                                                                                           | Une fois que   vous avez mis votre CV à jour et que vous êtes satisfait des expériences que   vous avez ajoutées, vous les utiliserez dans d’autres étapes pour nous aider   à mieux comprendre comment vous répondez aux exigences de compétences pour   cette occasion.                                                                                                                                                       | Application   step to begin working on profile, description, paragraph three                                     |
| Review   your résumé                                                                                                                                                                                                                                                                                                                                       | Passez en revue   votre CV                                                                                                                                                                                                                                                                                                                                                                                                      | Application   step to review your resume, title                                                                  |
| This   step allows you to edit any résumé information you’ve already added to your   profile. Click on an item to expand it, revealing more details. If you   haven’t added anything to your résumé yet, you can do so from this page by   selecting the “Add a new experience” button.                                                                    | Cette étape   vous permet de modifier toute information sur votre CV que vous avez déjà   ajoutée à votre profil. Cliquez sur un élément pour le développer, révélant   plus de détails. Si vous n’avez encore rien ajouté à votre CV, vous pouvez le   faire à partir de cette page en sélectionnant le bouton « Ajouter   une nouvelle expérience ».                                                                          | Application   step to review your resume, description                                                            |
| Create   your résumé                                                                                                                                                                                                                                                                                                                                       | Créez votre CV                                                                                                                                                                                                                                                                                                                                                                                                                  | Application   step to create your resume, title                                                                  |
| Creating   your résumé is a little different on this platform. First and foremost, any   work you do here will be saved to your profile so that you can reuse it on   other applications down the road.                                                                                                                                                    | La création de   votre CV est un peu différente sur cette plateforme. Tout d’abord, tout   travail que vous effectuez ici sera enregistré dans votre profil afin que   vous puissiez le réutiliser sur d’autres demandes ultérieurement.                                                                                                                                                                                        | Application   step to create your resume, description, paragraph one                                             |
| Building   your résumé consists of describing work experiences, education, community   participation, personal learning, and awards you’ve earned. In a later step,   you’ll use these experiences to highlight your skills. You can start adding   experiences to your résumé using the “Add a new experience” button.                                    | La rédaction de   votre CV consiste à décrire les expériences de travail, l’éducation, la   participation communautaire, l’apprentissage personnel et les récompenses que   vous avez gagnées. Dans une étape ultérieure, vous utiliserez ces expériences   pour mettre en valeur vos compétences. Vous pouvez commencer à ajouter des   expériences à votre CV en utilisant le bouton « Ajouter une nouvelle   expérience ».   | Application   step to create your resume, description, paragraph two                                             |
| I’m   happy with my résumé                                                                                                                                                                                                                                                                                                                                 | Je suis   satisfait des renseignements dans mon CV                                                                                                                                                                                                                                                                                                                                                                              | Action button   to move to the next step after reviewing the résumé                                              |
| Update   and review your résumé information.                                                                                                                                                                                                                                                                                                               | Mettez à jour   et révisez les renseignements de votre CV                                                                                                                                                                                                                                                                                                                                                                       | Application   step to review your resume, add an experience, page subtitle                                       |
| Add an   experience to your résumé                                                                                                                                                                                                                                                                                                                         | Ajoutez une   expérience à votre CV                                                                                                                                                                                                                                                                                                                                                                                             

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [x] This page exists at the proper route (behind feature flag)
- [x] On this page, page subtitle and breadcrumbs appear correctly
- [ ] The inner contents of the page (so not the hero and stepper layout) have a storybook story
- [x] Text should be different if you had no experiences when landing on this page
- [x] Experiences can be sorted by date or by type
- [x] The accordion for each experience type follows the look in the figma designs
- [x] Add new experience and Edit are links
- [x] "I'm happy with my resume" updates the Application to save that this step has submitted, than continues to next step if successful
- [x] "I'm happy with my resume" fails if user does not have at least one experience
- [x] Save and quit for now is a link back to dashboard

## 🛑 Blockers

Blocked by #6027, #5985 (these are soft blocks)

**Comment by esizer** [comment]
> I was looking at this issue earlier and found it seemed like a lot was going on, almost overwhelming. Felt like refactoring accordions should be spun off into its own thing rather than slipped into another issue given they are complex components.

Agreed, I went through a similar (but less impactful) issue when creating the add experience page. It seems there were changes/additions to components that slipped through the cracks when we were planning this epic.

See: https://github.com/GCTC-NTGC/gc-digital-talent/pull/6171#:~:text=Added%20a%20%60%20component%20to%20make%20making%20accessible%2C%20reusable%2C%20styled%20definition%20lists

**Comment by vd1992** [comment]
Will need to rename some fields, as well as make skills info an optional part of the accordion or remove it all together?

**Comment by yonikid15** [comment]
For the edit button we could add dialog trigger that overlaps the accordion trigger using some CSS positioning magic ✨. It would be nested within the header (h2), and next to the button. I tested it on the browser quickly and it looks like it could work?
![image](https://user-images.githubusercontent.com/22059495/229914783-12a81cdf-eb56-43e8-aed0-8a29f98a301e.png)

**Comment by esizer** [comment]
> For the edit button we could add dialog trigger that overlaps the accordion trigger using some CSS positioning magic ✨. It would be nested within the header (h2), and next to the button. I tested it on the browser quickly and it looks like it could work?

😆 While this works, it is a kind of an anti-pattern. This will pass automated tests and maybe a few manual tests but I don't think this is something we want to do because while it passes the spec, it still introduces issues for users with motor disabilities.

REF: [Overlapping interactive elements](https://www.tempertemper.net/blog/overlapping-interactive-areas#:~:text=Just%20because%20you%20can%20beat%20him%20up%2C%20doesn%E2%80%99t%20give%20you%20the%20right%20to.%20Remember%3A%20with%20great%20power%20comes%20great%20responsibility.).

**Comment by yonikid15** [comment]
> REF: [Overlapping interactive elements](https://www.tempertemper.net/blog/overlapping-interactive-areas#:~:text=Just%20because%20you%20can%20beat%20him%20up%2C%20doesn%E2%80%99t%20give%20you%20the%20right%20to.%20Remember%3A%20with%20great%20power%20comes%20great%20responsibility.).

We could also place the edit button after the `<Accordion.Trigger />` and expand the padding? This could work, but I'm not 100% sure it isn't an accessibility issue. Also, we would need approval from Josh about the design change.

![image](https://user-images.githubusercontent.com/22059495/229928104-1ab2917a-e726-4675-95de-2317a333af26.png)

**Comment by substrae** [comment]
If we can't overlap the interactive elements, I think @yonikid15's latest suggestion is the best option, where the trigger consumes the majority of the space and the edit link is adjacent, while looking like a part of the whole.

My concern with moving the edit link elsewhere is that it's not immediately obvious that it exists unless it is visible in the accordion's default state - so for example, you wouldn't know you could edit an experience unless you expanded it, which I know people will miss.

@esizer @yonikid15 can we also make sure that the title element in these accordions are underlined? The arrow, IMO, isn't creating enough affordance on its own, and having the title appear as clickable helps.
